pkg/costmodel/allocation_helpers.go

```diff
@@ -749,10 +751,26 @@ func applyNetworkAllocation(podMap map[podKey]*pod, resNetworkGiB []*prom.QueryR
 			for _, alloc := range thisPod.Allocations {
 				gib := res.Values[0].Value / float64(len(thisPod.Allocations))
 				costPerGiB := costPerGiBByCluster[podKey.Cluster]
-				alloc.NetworkCost = gib * costPerGiB / float64(len(pods))
+				currentNetworkSubCost := gib * costPerGiB / float64(len(pods))
+				switch networkCostSubType {
+				case networkZoneCost:
+					alloc.NetworkZoneCost = currentNetworkSubCost
+				case networkRegionCost:
+					alloc.NetworkRegionCost = currentNetworkSubCost
+				case networkInternetCost:
+					alloc.NetworkInternetCost = currentNetworkSubCost
+				default:
+					if !slices.Contains(unknownNetworkTypes, networkCostSubType) {
+						unknownNetworkTypes = append(unknownNetworkTypes, networkCostSubType)
+					}
```

Comment:
No need to collect these this way -- let's just log here directly and move forward. As written, we'll never end up here, anyways, as we control the only call to this private function. (Bigger deal if this were a public function.)

Reply:
addressed!